lib/active_merchant/billing/gateways/authorize_net_cim.rb

```diff
@@ -815,24 +815,24 @@ def add_credit_card(xml, credit_card)
       # Note: This element should only be included
       # when the payment method is bank account.
       def add_bank_account(xml, bank_account)
```

Comment:
This method currently accepts a Hash. Switching it to an ActiveMerchant::Billing::Check object will cause code that relies on this to break on an upgrade of the ActiveMerchant gem. My suggestion is that you modify the method to accept both the existing Hash and the new ActiveMerchant::Billing::Check object. If it's a Hash a Deprecation Warning to be displayed that in a future release Hash will no longer be supported.